lib/chef/api_client/registration.rb

```diff
@@ -102,10 +104,31 @@ def update
           @server_generated_private_key = response.private_key
         else # Chef 10
           @server_generated_private_key = response["private_key"]
+          response = api_client(response)
```

Comment:
should this be outside the if statement? Otherwise, update's return type is not consistent?